lib/cocoapods/installer/analyzer.rb

```diff
@@ -367,8 +367,8 @@ def analyze_host_targets_in_podfile(aggregate_targets, embedded_aggregate_target
       #
       def generate_targets
         specs_by_target = result.specs_by_target.reject { |td, _| td.abstract? }
-        check_pod_target_swift_versions(specs_by_target)
         pod_targets = generate_pod_targets(specs_by_target)
+        check_pod_target_swift_versions(pod_targets)
```

Comment:
@DanToml I know you've had some reservations about this but I don't think it breaks anything in terms of running the check after pod targets have been created.